Logger/FullName.cs

```diff
@@ -0,0 +1,15 @@
+﻿namespace Logger
+{
+    //The FullName is a value type that could be modified through copies when called upon
+    //while keeping the main copy unchanged.
+    //The FullName record will be immutable since the changes will be performed on copies.
+    public record struct FullName(string FirstName, string LastName, string? MiddleName = null)
+    {
+        
+        public string FirstName { get; } = FirstName ?? throw new ArgumentNullException(nameof(FirstName));
+        public string LastName { get; } = LastName ?? throw new ArgumentNullException(nameof(LastName));
+        public string MiddleName { get; } = MiddleName;
```

Comment:
```suggestion
        public string? MiddleName { get; } = MiddleName;
```
MiddleName should be marked as nullable - since that's its default value. This also gets rid of the nullable warning.

Reply:
We are straight-up lifting this recommendation and including in the PR.